### PR TITLE
net-p2p/litecoind: add 0.18.1

### PR DESCRIPTION
**Pull request by SoapGentoo**
Closes: https://bugs.gentoo.org/607842
Bug: https://bugs.gentoo.org/672326
Bug: https://bugs.gentoo.org/768768
Bug: https://bugs.gentoo.org/788844
Signed-off-by: David Seifert <soap@gentoo.org>

### CODE REVIEW

File: net-p2p/litecoind/litecoind-0.18.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+DESCRIPTION="The offical daemon to run your own (full) Litecoin node"
+HOMEPAGE="https://litecoin.org/"
+SRC_URI="https://github.com/${MY_PN}-project/${MY_PN}/archive/v${MY_PV}.tar.gz -> ${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="MIT ISC GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="logrotate upnp +wallet"
```
Comment (thesamesam) [inline net-p2p/litecoind/litecoind-0.18.1.ebuild:22]:
```suggestion
IUSE="upnp +wallet"
```
and remove src_install usage

**Comment by SoapGentoo** [comment]
```sh
--- litecoind-0.10.4.0-r2.ebuild
+++ litecoind-0.18.1.ebuild
@@ -26,44 +26,30 @@
 	acct-user/litecoin
 	dev-libs/boost:=[threads(+)]
 	dev-libs/leveldb:=
+	dev-libs/libevent:=[threads(+)]
 	dev-libs/openssl:0[-bindist(-)]
 	sys-libs/db:$(db_ver_to_slot "${DB_VER}")[cxx]
 	logrotate? ( app-admin/logrotate )
-	upnp? ( net-libs/miniupnpc )
-"
-DEPEND="
-	${RDEPEND}
-	>=app-shells/bash-4.1
-	sys-apps/sed
-"
-
-PATCHES=(
-	"${FILESDIR}"/0.9.0-sys_leveldb.patch
-	"${FILESDIR}"/litecoind-0.10.2.2-memenv_h.patch
-	"${FILESDIR}"/litecoind-0.10.2.2-fix-gnustack.patch
-	"${FILESDIR}"/${P}-gcc6.patch
-)
+	upnp? ( net-libs/miniupnpc )"
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}"/${P}-system-leveldb.patch )
 
 src_prepare() {
 	default
 
 	eautoreconf
-	rm -r src/leveldb
+	rm -r src/leveldb || die
 }
 
 src_configure() {
 	# To avoid executable GNU stack.
 	append-ldflags -Wl,-z,noexecstack
 
-	local my_econf=
-	if use upnp; then
-		my_econf="${my_econf} --with-miniupnpc --enable-upnp-default"
-	else
-		my_econf="${my_econf} --without-miniupnpc --disable-upnp-default"
-	fi
-
 	local myeconfargs=(
 		$(use_enable wallet)
+		$(use_with upnp miniupnpc)
+		$(use_enable upnp upnp-default)
 		--disable-ccache
 		--disable-static
 		--disable-tests
@@ -73,9 +59,7 @@
 		--with-daemon
 		--without-gui
 		--without-qrencode
-		${my_econf}
 	)
-
 	econf "${myeconfargs[@]}"
 }
 
@@ -83,13 +67,13 @@
 	default
 
 	insinto /etc/litecoin
-	doins "${FILESDIR}/litecoin.conf"
+	doins "${FILESDIR}"/litecoin.conf
 	fowners litecoin:litecoin /etc/litecoin/litecoin.conf
 	fperms 600 /etc/litecoin/litecoin.conf
 
-	newconfd "${FILESDIR}/litecoin.confd" ${PN}
-	newinitd "${FILESDIR}/litecoin.initd-r1" ${PN}
-	systemd_dounit "${FILESDIR}/litecoin.service"
+	newconfd "${FILESDIR}"/litecoin.confd ${PN}
+	newinitd "${FILESDIR}"/litecoin.initd-r1 ${PN}
+	systemd_dounit "${FILESDIR}"/litecoin.service
 
 	keepdir /var/lib/litecoin/.litecoin
 	fperms 700 /var/lib/litecoin
@@ -98,11 +82,9 @@
 	dosym /etc/litecoin/litecoin.conf /var/lib/litecoin/.litecoin/litecoin.conf
 
 	dodoc doc/README.md doc/release-notes.md
-	newman contrib/debian/manpages/bitcoind.1 litecoind.1
-	newman contrib/debian/manpages/bitcoin.conf.5 litecoin.conf.5
 
 	if use logrotate; then
 		insinto /etc/logrotate.d
-		newins "${FILESDIR}/litecoind.logrotate" litecoind
+		newins "${FILESDIR}"/litecoind.logrotate litecoind
 	fi
 }
```

File: net-p2p/litecoind/litecoind-0.18.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	# To avoid executable GNU stack.
+	append-ldflags -Wl,-z,noexecstack
+
+	local myeconfargs=(
+		$(use_enable wallet)
+		$(use_with upnp miniupnpc)
+		$(use_enable upnp upnp-default)
+		--disable-ccache
+		--disable-static
+		--disable-tests
```
Comment (thesamesam) [inline net-p2p/litecoind/litecoind-0.18.1.ebuild:55]: Any chance of getting them working? (Fine if not given this is completely unbuildable atm)

File: net-p2p/litecoind/litecoind-0.18.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+RDEPEND="
+	acct-group/litecoin
+	acct-user/litecoin
+	dev-libs/boost:=[threads(+)]
+	dev-libs/leveldb:=
+	dev-libs/libevent:=[threads(+)]
+	dev-libs/openssl:0[-bindist(-)]
+	sys-libs/db:$(db_ver_to_slot "${DB_VER}")[cxx]
+	logrotate? ( app-admin/logrotate )
```
Comment (thesamesam) [inline net-p2p/litecoind/litecoind-0.18.1.ebuild:32]:
```suggestion
```

File: net-p2p/litecoind/litecoind-0.18.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	upnp? ( net-libs/miniupnpc )"
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}"/${P}-system-leveldb.patch )
+
+src_prepare() {
+	default
+
+	eautoreconf
+	rm -r src/leveldb || die
```
Comment (thesamesam) [inline net-p2p/litecoind/litecoind-0.18.1.ebuild:42]: It's a bit gross being below euatoreconf but I get it (may need a configure patch otherwise)

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Review by thesamesam — APPROVED** [review]

File: net-p2p/litecoind/litecoind-0.18.1.ebuild (hunk 2 above)
Comment (SoapGentoo) [reply net-p2p/litecoind/litecoind-0.18.1.ebuild:54]: nope, segfaults and need to comment out more tests

**Pull request closed without merging**